test/IECoreScene/MeshAlgoTriangulateTest.py

```diff
@@ -0,0 +1,297 @@
+
+		m = IECoreScene.MeshPrimitive( verticesPerFace, vertexIds )
+		m["P"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Vertex, P )
+
+		def testTriangulate():
+			IECoreScene.MeshAlgo.triangulate( m, throwExceptions = True )
+
+		# FloatVectorData not valid for "P"
+		self.assertRaises( RuntimeError, testTriangulate )
+
+	def testConstantPrimVars( self ) :
+
+		m = IECore.Reader.create( "test/IECore/data/cobFiles/polySphereQuads.cob").read()
+
+		m["constantScalar"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Constant, IECore.FloatData( 1 ) )
+		m["constantArray"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Constant, IECore.StringVectorData( [ "one", "two" ] ) )
+
+		result = IECoreScene.MeshAlgo.triangulate( m )
+		self.assert_( result.arePrimitiveVariablesValid() )
+
+	def testInterpolationShouldntChange( self ) :
+
+		m = IECoreScene.MeshPrimitive.createPlane( imath.Box2f( imath.V2f( -1 ), imath.V2f( 1 ) ) )
+		m.setTopology( m.verticesPerFace, m.vertexIds, "catmullClark" )
+
+		m2 = IECoreScene.MeshAlgo.triangulate( m )
+
+		self.assertEqual( m2.interpolation, "catmullClark" )
+
+	def testFaceVaryingIndices( self ) :
+
+		m = IECoreScene.MeshPrimitive.createPlane( imath.Box2f( imath.V2f( -1 ), imath.V2f( 1 ) ) )
+		m["uv"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.FaceVarying, m["uv"].data, IECore.IntVectorData( [ 0, 3, 1, 2 ] ) )
+
+		m2 = IECoreScene.MeshAlgo.triangulate( m )
+
+		self.assertTrue( m2.arePrimitiveVariablesValid() )
+
+		self.assertEqual( m2["uv"].data, m["uv"].data )
+		self.assertEqual( m2["uv"].indices, IECore.IntVectorData( [ 0, 3, 1, 0, 1, 2 ] ) )
+
+	def testUniformIndices( self ) :
+
+		m = IECoreScene.MeshPrimitive.createPlane( imath.Box2f( imath.V2f( -4 ), imath.V2f( 4 ) ), divisions = imath.V2i( 2, 2 ) )
+		m["myString"] = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Uniform, IECore.StringVectorData( [ "a", "b" ] ), IECore.IntVectorData( [ 1, 0, 0, 1 ] ) )
+
+		m2 = IECoreScene.MeshAlgo.triangulate( m )
+
+		self.assertTrue( m2.arePrimitiveVariablesValid() )
+
+		self.assertEqual( m2["myString"].data, m["myString"].data )
+		self.assertEqual( m2["myString"].indices, IECore.IntVectorData( [ 1, 1, 0, 0, 0, 0, 1, 1 ] ) )
+
+	@unittest.skipUnless( os.environ.get("CORTEX_PERFORMANCE_TEST", False), "'CORTEX_PERFORMANCE_TEST' env var not set" )
+	def testTriangulatePerformance( self ):
+
+		for i in range(10):
+			root = IECoreScene.SceneInterface.create( "/path/to/cache.scc", IECore.IndexedIO.OpenMode.Read )
+			body = root.scene(['location', 'of', 'object'])
+
+			objects = []
+
+			for f in range( 850, 1000 ):
+				objects.append( body.readObject(f / 24.0 ))
+
+			timer = IECore.Timer( True, IECore.Timer.Mode.WallClock )
+
+			totalNumTriangles = 0
+			for o in objects:
+				o = IECoreScene.TriangulateOp()( input = o, throwExceptions = False )
```

Comment:
we should be using the algo for the performance test, right?

Reply:
552ba76